sklearn/tree/export.py

```diff
@@ -55,32 +100,191 @@ def export_graphviz(decision_tree, out_file="tree.dot", feature_names=None,
     >>> tree.export_graphviz(clf,
     ...     out_file='tree.dot')                # doctest: +SKIP
     """
-    def node_to_str(tree, node_id, criterion):
-        if not isinstance(criterion, six.string_types):
-            criterion = "impurity"
 
-        value = tree.value[node_id]
-        if tree.n_outputs == 1:
-            value = value[0, :]
-
-        if tree.children_left[node_id] == _tree.TREE_LEAF:
-            return "%s = %.4f\\nsamples = %s\\nvalue = %s" \
-                   % (criterion,
-                      tree.impurity[node_id],
-                      tree.n_node_samples[node_id],
-                      value)
+    def colour_brew(n):
```

Comment:
General comment: I am not sure we should define all these auxiliary functions within `export_graphviz`. I think it would be more pythonic to make them top-level, but private (e.g., `_colour_brew`). What do you think? @amueller